cmd/snap/cmd_snapshot.go

```diff
@@ -0,0 +1,297 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/strutil"
+	"github.com/snapcore/snapd/strutil/quantity"
+)
+
+func fmtSize(size int64) string {
+	return quantity.FormatAmount(uint64(size), -1)
+}
+
+type savedCmd struct {
+	clientMixin
+	durationMixin
+	ID         snapshotID `long:"id"`
+	Positional struct {
+		Snaps []installedSnapName `positional-arg-name:"<snap>"`
+	} `positional-args:"yes"`
+}
+
+var shortSavedHelp = i18n.G("List currently stored snapshots")
+var longSavedHelp = i18n.G(`
+The saved command lists the snapshots that have been created previously with
+the 'save' command.
+`)
+
+func (x *savedCmd) Execute([]string) error {
+	list, err := x.client.SnapshotSets(uint64(x.ID), installedSnapNames(x.Positional.Snaps))
+	if err != nil {
+		return err
+	}
+	if len(list) == 0 {
+		fmt.Fprintln(Stdout, i18n.G("No snapshots found."))
+		return nil
+	}
+	w := tabWriter()
+	defer w.Flush()
+
+	fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\t%s\n",
+		// TRANSLATORS: 'Set' as in group or bag of things
+		i18n.G("Set"),
+		"Snap",
+		i18n.G("Age"),
```

Comment:
`# Translators: as in how old something is, not as in the stone age`

Reply:
good catch